homeassistant/components/iometer/sensor.py

```diff
@@ -0,0 +1,160 @@
+"""IOmeter sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS,
+    STATE_UNKNOWN,
+    EntityCategory,
+    UnitOfEnergy,
+    UnitOfPower,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .coordinator import IOMeterCoordinator, IOmeterData
+from .entity import IOmeterEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class IOmeterEntityDescription(SensorEntityDescription):
+    """Describes IOmeter sensor entity."""
+
+    value_fn: Callable[[IOmeterData], str | int | float]
+
+
+SENSOR_TYPES: list[IOmeterEntityDescription] = [
+    IOmeterEntityDescription(
+        key="meter_number",
+        translation_key="meter_number",
+        icon="mdi:meter-electric",
+        value_fn=lambda data: data.status.meter.number,
+    ),
+    IOmeterEntityDescription(
+        key="connection_status",
+        translation_key="connection_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=["connected", "disconnected"],
```

Comment:
should this one be a binary_sensor instead? 
(if so, we should add it in a separate PR since we should have only 1 platform per PR)

Reply:
Yes, we could do that. How would the EntityDescription look like?

Reply:
Okay, I removed it and will add it in a future PR